semgrep-core/parsing/Parse_kotlin_tree_sitter.ml

```diff
@@ -1358,12 +1366,13 @@ and primary_constructor (env : env) ((v1, v2) : CST.primary_constructor) =
   let v2 = class_parameters env v2 in
   todo env (v1, v2)
 
-and primary_expression (env : env) (x : CST.primary_expression) : AST.expr =
+and primary_expression (env : env) (x : CST.primary_expression) : expr =
   (match x with
-  | `Paren_exp x -> let _ = parenthesized_expression env x in
-      raise Todo
-  | `Simple_id x -> simple_identifier env x
-  | `Lit_cst x -> literal_constant env x
+  | `Paren_exp x -> parenthesized_expression env x
+  | `Simple_id x -> 
+      let id = simple_identifier env x in
+      Id(id, empty_id_info())
+  | `Lit_cst x -> pr2 "in lit_cst"; L (literal_constant env x)
```

Comment:
there's a pr2 here.
We actually should have a semgrep rule preventing to commit such code.

Reply:
oh shoot, I'll change that! and very true we should have a semgrep rule for this

Reply:
I will write this this week or next 🤔